packages/puppeteer-core/src/api/HTTPRequest.ts

```diff
@@ -548,6 +548,29 @@ export abstract class HTTPRequest {
       return;
     }
   }
+
+  /**
+   * @internal
+   */
+  static getResponse(body: string | Uint8Array): {
+    contentLength: number;
+    base64: string;
+  } {
+    // Needed to get the correct byteLength
+    const byteBody: Uint8Array = isString(body)
```

Comment:
Let's add a unit test comparing the results of this helper with Buffer implementation: let's try empty string, ' ', multi-byte character, latin characters. It looks equivalent but just to be sure.